src/sentry/static/sentry/app/views/organizationIntegrationConfig.jsx

```diff
@@ -0,0 +1,305 @@
+  color: ${p => p.theme.gray3};
+  font-size: 14px;
+`;
+
+const Details = styled(Flex)`
+  font-size: 15px;
+  line-height: 2.1rem;
+`;
+
+const Description = styled.div`
+  li {
+    margin-bottom: 6px;
+  }
+`;
+
+const AuthorName = styled.div`
+  color: ${p => p.theme.gray2};
+`;
+
+const MetadataLink = styled.a`
+  display: block;
+  margin-bottom: 6px;
+`;
+
+const alertLinkMarkedRenderer = new marked.Renderer();
+alertLinkMarkedRenderer.paragraph = s => s;
+
+function computeCenteredWindow(width, height) {
+  const screenLeft = window.screenLeft != undefined ? window.screenLeft : screen.left;
+  const screenTop = window.screenTop != undefined ? window.screenTop : screen.top;
+  const innerWidth = window.innerWidth
+    ? window.innerWidth
+    : document.documentElement.clientWidth
+      ? document.documentElement.clientWidth
+      : screen.width;
+
+  const innerHeight = window.innerHeight
+    ? window.innerHeight
+    : document.documentElement.clientHeight
+      ? document.documentElement.clientHeight
+      : screen.height;
+
+  const left = innerWidth / 2 - width / 2 + screenLeft;
+  const top = innerHeight / 2 - height / 2 + screenTop;
+
+  return {left, top};
+}
+
+export default class OrganizationIntegrationConfig extends AsyncView {
+  componentDidMount() {
+    this.dialog = null;
+    window.addEventListener('message', this.receiveMessage, false);
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount();
+    window.removeEventListener('message', this.receiveMessage);
+
+    if (this.dialog !== null) {
+      this.dialog.close();
+    }
+  }
+
+  getProvider() {
+    const {config} = this.state;
+
+    if (config !== null) {
+      return config.providers.find(p => p.key == this.props.params.providerKey) || null;
+    }
+
+    return null;
+  }
+
+  getTitle() {
+    const provider = this.getProvider();
+
+    if (provider === null) {
+      return 'Global Integrations';
+    }
+
+    return `${provider.name} Integration`;
+  }
+
+  getEndpoints() {
+    const {orgId} = this.props.params;
+
+    return [
+      ['config', `/organizations/${orgId}/config/integrations/`],
+      ['itemList', `/organizations/${orgId}/integrations/`, {query: {status: ''}}],
+    ];
+  }
+
+  handleAddIntegration = provider => {
+    const name = 'sentryAddIntegration';
+
+    const {url, width, height} = provider.setupDialog;
+    const {left, top} = computeCenteredWindow(width, height);
+
+    this.dialog = window.open(
+      url,
+      name,
+      `scrollbars=yes,width=${width},height=${height},top=${top},left=${left}`
+    );
+
+    this.dialog.focus();
+    this.dialog.onclose = () => document.location.refresh();
+  };
+
+  receiveMessage = message => {
+    if (message.origin !== document.origin) {
+      return;
+    }
+
+    if (message.source !== this.dialog) {
+      return;
+    }
+
+    this.dialog = null;
+
+    const {success, data} = message.data;
+
+    if (!success) {
+      // TODO: Error state here
```

Comment:
when would this happen? at the very least capture this for raven? and/or add a failure indicator?

Reply:
Looks like this doesn't happen at all now. I'm just going to add a failed indicator